lib/matplotlib/axes/_axes.py

```diff
@@ -4014,14 +4014,15 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
         else:
             try:
                 c_array = np.asanyarray(c, dtype=float)
-                if c_array.size == x.size:
```

Comment:
The docstrings say x/y must be `(n, )` shaped, but it does not look like we check that anywhere.  I suspect we have the ravel to in fact support `(n, 1)` and `(1, n)` shaped arrays arrays.